python/cudf/cudf/core/series.py

```diff
@@ -2895,6 +2898,9 @@ def product(
                 "numeric_only parameter is not implemented yet"
             )
 
+        if isinstance(self._column, (DatetimeColumn, TimeDeltaColumn)):
+            raise TypeError(f"cannot perform prod with type {self.dtype}")
```

Comment:
We really shouldn't have to check these in the `Series` call. They should be checked when calling into the `column.product()` call. Maybe need to push some of the additional logic here down?